### ISSUE TITLE
SPFx web part won't populate DOM (on first load) in Viva Connections Desktop

### ISSUE THREAD

**Issue by poneilcsgp** [SharePoint/sp-dev-docs#7971]
I've created a simple web part for use on our corporate intranet that uses react-autosuggest to provide suggested names (based on graph query of users) to pass to a search page (via querysting).    The part works fine when the SharePoint site is loaded in the browser and if you view the site as a website tab in a Team.   

We have used the Viva Connections Desktop app to make our intranet available from within teams.   The issue is that the web part will not show the react-autosuggest suggestions after the page first loads, but if I click the "search button" the web part passes the search to the same page (via querystring params) and then the autosuggest works fine (until I leave the Viva Connections app and go to somewhere else in teams).   The behavior is the same in the browser version of Teams.  For some reason it just won't  generate the DOM elements on first load (I have console logs showing me that the graph search is happening/data coming back).

Totally stumped on this one because it's really hard to test when it works everywhere except in Viva Connections app.

**Comment by patmill** [comment]
Have you set up any breakpoints to see where the issue is getting stalled?  For example, you mention "it won't generate the DOM elements".  What DOM element?  Is it an extension placeholder?  The webpart div?  Is your render method getting called?  Is your onInit function returning a promise that resolves?

**Comment by poneilcsgp** [comment]
Sorry for the delay. I'm using the react-autosuggest component to provide autosuggest functionality when typing.  After a set of suggestions is fetched it renders them in a DIV that's part of the autosuggest component. When loading on the SharePoint page directly in Chrome, edge, safari, etc. and typing in some text the graph api is called, returns results and I set them to the autosuggest state and the div is populated with the suggestions (by autosuggest).   Running locally and testing in SPO workbench the component works fine.

In Teams with the Dev tools I can see that the graph api is called, but the div is not populated.   

FWIW I've also tried to use the SPO workbench in teams (by using the Dev Tools to change window location) but even though standard webparts load fine these Webparts don't load at all.  I'm not sure if Workbench in a Teams app is supposed to work, but I figured I'd give it a shot since getting representative testing locally would help a lot.

I'm back to this now and inserting a bunch of logging to see if I can figure out where it's not firing.

**Comment by patmill** [comment]
Where is the div for the autosuggest component located?  Is it inside the webpart's div?

**Comment by poneilcsgp** [comment]
See attached.   The suggestions that appear below the textbox in the 'react-autowhatever-1' div.
<img width="530" alt="image" src="https://user-images.githubusercontent.com/104784418/170700400-0c485208-888d-4f1a-a359-052d2f410655.png">

Div for the suggestions is right under the textbox.   You can see in the (shadow) DOM:

<img width="449" alt="image" src="https://user-images.githubusercontent.com/104784418/170700811-018b5cf2-7b4f-4cd4-9882-127934dd434d.png">

When I do the same thing in Teams when the component loads without any other interaction, the suggestions html do not appear in the DIV, but if I do a Search (search button just redirects to the same page adding search values to query string parameters, the suggestions will start to appear.    I was trying to figure out if it was a package size or lazy loading issue with Teams (where some code is forced to load when search button is clicked).  As I mentioned earlier, works totally fine when accessed via browser directly in SharePoint.

**Comment by poneilcsgp** [comment]
Another followup.  Since I couldn't get workbench testing to work in teams, I started debugging using this guidance:  https://www.eliostruyf.com/testing-and-debugging-your-spfx-solutions-in-production-without-causing-any-impact/ by opening dev tools in Teams and doing window.location.href = window.location.href + '&loadSPFX=true&debugManifestsFile=https://localhost:4321/temp/manifests.js' from the DevTools console.    This confirms that Autosuggest component doesn't fire the rendersuggestions when using the component in Teams.  

It feels like I've got better ways to test, but I don't yet have insight into 

1.   Why it renderSuggestions doesn't run on first load in Viva Connections
2.  Why it DOES work after the webpart does a simple redirect to the same page with some search results and then works fine on any page thereafter.
3.  Why it has no issues in SP in browser.

**Comment by poneilcsgp** [comment]
Ok, I think I've got it figured out.  Not 100% sure why it was ok in Browser or started working after first submit, but it looks like it was related to how what the state was being set to in instances when nothing was coming from query string params.  I wrapped some additional logic around the first load and it appears to work in Teams and SP now.      I'll update again if there's something that suggests otherwise.  Thank you a ton for your assistance, helped me to re-scrutinize my code vs being confused by the Teams/Viva vs. Browser/SP scenario.
